fipy/solvers/__init__.py

```diff
@@ -128,3 +136,8 @@ def __init__(self, solver):
                  test=lambda: solver == 'pysparse',
                  why="the PySparse solvers are not being used.",
                  skipWarning=True)
+
+register_skipper(flag='PYAMGX_SOLVER',
```

Comment:
The sematics for this skipper are backwards.  The other skippers, e.g. `SCIPY_SOLVER`, are for when the item *is* being used, not when it's not.

Simplest fix is rename this to `NOT_PYAMGX_SOLVER`.